### PR TITLE
8263024: Convert Valhalla tests using the old framework to the new framework

### PR DESCRIPTION
**Pull request by katyapav**
This PR converts compiler/valhalla/inlinetypes tests based on old Valhalla/IR test framework to new one 
which was recently integrated by Christian as part of JDK-8254129.
The tests which were we decided to convert are basically the ones which extends InlineTypeTest class.
The rest of compiler/valhalla/inlinetypes tests were agreed (with Christian and Tobias) to keep untouched 
as they are in most cases regression tests.

The conversion rules/approach is pretty well described in the bug's description section (see JDK-8263024).
A detailed description of new IR test framework could found in test/hotspot/jtreg/compiler/lib/ir_framework/README.md.

Testing:
 Ran compiler/valhalla/inlinetypes in all the configurations used in hs-tier1-9.
 There 2 failed converted tests:
  compiler/valhalla/inlinetypes//TestNullableArrays.java - new bug JDK-8269070 filed
  compiler/valhalla/inlinetypes/TestIntrinsics.java - know issue tracked by JDK-8239003

Big thanks to Christian and Tobias for advice and help during this tests conversion.

Please review the changes.

Thanks,
Katya

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8263024](https://bugs.openjdk.java.net/browse/JDK-8263024): Convert Valhalla tests using the old framework to the new framework

### Reviewers
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - Committer) ⚠️ Review applies to 63bdc6f2998ac3a943ed27eb6030858c58199c13
 * [Christian Hagedorn](https://openjdk.java.net/census#chagedorn) (@chhagedorn - no project role) ⚠️ Review applies to 1e1dbe1dba1c52ff7f0201b658c94b0cb3848294

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/valhalla pull/457/head:pull/457` \
`$ git checkout pull/457`

Update a local copy of the PR: \
`$ git checkout pull/457` \
`$ git pull https://git.openjdk.java.net/valhalla pull/457/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 457`

View PR using the GUI difftool: \
`$ git pr show -t 457`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/valhalla/pull/457.diff">https://git.openjdk.java.net/valhalla/pull/457.diff</a>

</details>

### CODE REVIEW

**Review by TobiHartmann — APPROVED** [review]
Thanks a lot for taking care of this, Katya! The changes look good to me, just TestIntrinsics.java needs merging.

Please link JDK-8263024 to the PR.

Best regards,
Tobias

**Comment by katyapav** [comment]
@TobiHartmann, thanks for your review!
I did merge TestIntrinsics.java and link the bug.

**Review by TobiHartmann — APPROVED** [review]
Thanks, that looks good to me.

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestNullableArrays.java [hunk 1]
```diff
@@ -97,22 +104,24 @@ public void test1_verifier(boolean warmup) {
...
+    public void test2_verifier() {
         long result = test2();
         Asserts.assertEQ(result, hash());
     }
 
     // Test receiving an inline type array from the interpreter,
     // updating its elements in a loop and computing a hash.
-    @Test(failOn = ALLOCA)
+    @Test
+    @IR(failOn = {ALLOCA})
```
Comment (chhagedorn) [inline test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestNullableArrays.java:124]: Can be simplified to `failOn = ALLOCA`.

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestNullableArrays.java [hunk 2]
```diff
@@ -2556,13 +2563,14 @@ public void test96_verifier(boolean warmup) {
     }
 
     // Test loads from vararg arrays
-    @Test(failOn = LOAD_UNKNOWN_INLINE)
+    @Test
+    @IR(failOn = {LOAD_UNKNOWN_INLINE})
```
Comment (chhagedorn) [inline test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestNullableArrays.java:2567]: Can be simplified to `failOn = LOAD_UNKNOWN_INLINE`.

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestUnloadedInlineTypeField.java [hunk 3]
```diff
@@ -22,50 +22,50 @@
...
- *                               -XX:PerMethodRecompilationCutoff=-1 -XX:PerBytecodeRecompilationCutoff=-1
- *                               compiler.valhalla.inlinetypes.InlineTypeTest
- *                               compiler.valhalla.inlinetypes.TestUnloadedInlineTypeField
+ * @run driver/timeout=300 compiler.valhalla.inlinetypes.TestUnloadedInlineTypeField
  */
 
-public class TestUnloadedInlineTypeField extends InlineTypeTest {
-    // Only prevent loading of classes when testing with C1. Load classes
+public class TestUnloadedInlineTypeField {
+    // The test only prevent loading of classes when testing with C1. Load classes
```
Comment (chhagedorn) [inline test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestUnloadedInlineTypeField.java:46]: `The test only` -> `Only`?

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestUnloadedInlineTypeField.java [hunk 4]
```diff
@@ -878,8 +878,8 @@ public MyValue20 test20() {
         return new MyValue20();
     }
 
-    @DontCompile
-    public void test20_verifier(boolean warmup) {
+    @Run(test = "test20")
+    public void test20_verifier(RunInfo info) {
```
Comment (chhagedorn) [inline test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestUnloadedInlineTypeField.java:882]: `RunInfo info` can be removed.

**Review by chhagedorn — CHANGES_REQUESTED** [review]
Thanks a lot for your help to do the conversions! They look good overall. There are only some minor things, mostly about code style.

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestLWorldProfiling.java [hunk 5]
```diff
@@ -461,415 +528,497 @@ public void test18_verifier(boolean warmup) {
...
         test20(testNotFlattenableArray, notFlattenable);
     }
 
     // acmp tests
 
     // branch frequency profiling causes not equal branch to be optimized out
-    @Warmup(10000)
-    @Test(failOn = SUBSTITUTABILITY_TEST)
+    @Test
+    @IR(failOn = {SUBSTITUTABILITY_TEST})
```
[the remark this answers is not in the document]
Comment (katyapav) [reply test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestLWorldProfiling.java:569]: I actually specially kept {} so it will be more straightforward to add more nodes if needed. But I can remove {} if you prefer simplified version.

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestNullableArrays.java (hunk 1 above)
Comment (katyapav) [reply test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestNullableArrays.java:124]: see previous comment

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestNullableArrays.java (hunk 2 above)
Comment (katyapav) [reply test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestNullableArrays.java:2567]: see previous comment

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestLWorld.java [hunk 6]
```diff
@@ -3769,14 +3852,14 @@ public int get() {
...
     }
 
-    @DontCompile
-    public void test141_verifier(boolean warmup) {
+    @Run(test = "test141")
+    @Warmup(0)
+    public void test141_verifier() {
         int result = test141();
         Asserts.assertEquals(result, 0);
     }
```
[the remark this answers is not in the document]
Comment (katyapav) [reply test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestLWorld.java:3865]: Good catch :), converted

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestIntrinsics.java [hunk 7]
```diff
@@ -82,7 +86,8 @@ public void test1_verifier(boolean warmup) {
     }
 
     // Verify that Class::isAssignableFrom checks with statically known classes are folded
-    @Test(failOn = LOADK)
+    @Test
+    @IR(failOn = {LOADK})
```
[the remark this answers is not in the document]
Comment (katyapav) [reply test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestIntrinsics.java:90]: I specially kept {} so it will be more straightforward to add more nodes if needed. But I can remove {} if you prefer simplified version.

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestIntrinsics.java [hunk 8]
```diff
@@ -108,16 +113,17 @@ public void test2_verifier(boolean warmup) {
...
         Asserts.assertTrue(test3(Object.class) == null, "test3_1 failed");
         Asserts.assertTrue(test3(MyValue1.ref.class) == MyAbstract.class, "test3_2 failed");
         Asserts.assertTrue(test3(MyValue1.val.class) == MyAbstract.class, "test3_3 failed");
         Asserts.assertTrue(test3(Class.class) == Object.class, "test3_4 failed");
     }
 
     // Verify that Class::getSuperclass checks with statically known classes are folded
-    @Test(failOn = LOADK)
+    @Test
+    @IR(failOn = {LOADK})
```
[the remark this answers is not in the document]
Comment (katyapav) [reply test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestIntrinsics.java:126]: I actually specially kept {} so it will be more straightforward to add more nodes if needed. But I can remove {} if you prefer simplified version.

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestIntrinsics.java [hunk 9]
```diff
@@ -374,55 +380,59 @@ public void test20_verifier(boolean warmup) {
 
     protected static final String CALL_Unsafe = START + "CallStaticJava" + MID + "# Static  jdk.internal.misc.Unsafe::" + END;
 
-    @Test(failOn=CALL_Unsafe)
+    @Test
+    @IR(failOn = {CALL_Unsafe})
```
[the remark this answers is not in the document]
Comment (katyapav) [reply test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestIntrinsics.java:384]: see previous comment

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestLWorld.java [hunk 10]
```diff
@@ -3389,33 +3461,35 @@ public void test123_verifier(boolean warmup) {
...
+    @Run(test = "test124")
+    public void test124_verifier() {
         test124(42, 42);
         test124(42, testValue1);
     }
 
     // acmp doesn't need substitutability test when one input null
-    @Test(failOn = SUBSTITUTABILITY_TEST)
+    @Test
+    @IR(failOn = {SUBSTITUTABILITY_TEST})
```
[the remark this answers is not in the document]
Comment (katyapav) [reply test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestLWorld.java:3478]: see previous comment

**Comment by katyapav** [comment]
I did rerun compiler/valhalla/inlinetypes in all the configurations used in hs-tier1-9
and only compiler/valhalla/inlinetypes/TestIntrinsics.java failed which is known issue tracked by JDK-8239003

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestCallingConventionC1.java [hunk 11]
```diff
@@ -173,17 +174,21 @@ static Functor getFunctor() {
     static class MyImplPojo1 implements Intf {
         int field = 1000;
 
+        @DontInline
         @ForceCompile(CompLevel.C1_SIMPLE)
         public int func1(int a, int b)             { return field + a + b + 20; }
```
Comment (chhagedorn) [inline test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestCallingConventionC1.java:179]: Can you also add a new line here?

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestCallingConventionC1.java [hunk 12]
```diff
@@ -173,17 +174,21 @@ static Functor getFunctor() {
...
         @ForceCompile(CompLevel.C1_SIMPLE)
         public int func2(int a, int b, Point p)    { return field + a + b + p.x + p.y + 20; }
     }
 
     static class MyImplPojo2 implements Intf {
         int field = 2000;
 
+        @DontInline
         @ForceCompile(CompLevel.C2)
         public int func1(int a, int b)             { return field + a + b + 20; }
```
Comment (chhagedorn) [inline test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestCallingConventionC1.java:190]: And here

**Review by chhagedorn — APPROVED** [review]
Thanks for doing the changes!
